lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -965,6 +965,86 @@ class MongoClientInterface {
         ], cb);
     }
 
+    /**
+     * gets object metadata for a list of objects
+     * @param {String} bucketName bucket name
+     * @param {Array} objects array of objects
+     * @param {Object} log logger
+     * @param {Function} callback callback
+     * @return {undefined}
+     */
+    getObjects(bucketName, objects, log, callback) {
+        const c = this.getCollection(bucketName);
+        // We do not accept more than 1000 keys in a single request
+        if (objects.length > 1000) {
```

Comment:
logically, should this check not be done after the next one (`Array.isArray`) ?

Reply:
yes, definitely: https://github.com/scality/Arsenal/pull/2134/commits/74218ad3ef820998d9970bfdb367e50f38608f25